probe-rs/src/debug/debug_info.rs

```diff
@@ -2084,4 +2129,60 @@ mod test {
 
         insta::assert_snapshot!(printed_backtrace);
     }
+    #[test]
+    fn probe_rs_debug_unwind_tests() {
+        // TODO: Add RISC-V tests.
+        for chip_name in ["nRF52833_xxAA", "RP2040"] {
+            let debug_info = debug_info(format!("debug-unwind-tests/{chip_name}.elf").as_str());
+            let coredump = fs::read(get_path_for_test_files(
+                format!("debug-unwind-tests/{chip_name}.coredump").as_str(),
+            ))
+            .unwrap();
+            let mut adapter = CoreDump::load_raw(&coredump).unwrap();
+
+            let initial_registers = adapter.debug_registers();
+            let exception_handler = exception_handler_for_core(adapter.core_type());
+            let instruction_set = adapter.instruction_set();
+
+            let mut stack_frames = debug_info
+                .unwind(
+                    &mut adapter,
+                    initial_registers,
+                    exception_handler.as_ref(),
+                    Some(instruction_set),
+                )
+                .unwrap();
+
+            let snapshot_name = format!("{chip_name}__full_unwind");
+
+            // Expand and validate the static and local variables for each stack frame.
+            for frame in stack_frames.iter_mut() {
+                for variable_cache in [
```

Comment:
This could be done with a `https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.flat_map` to simplify things possibly :)

Reply:
These for loops do not produce any new elements (iterators or other), they simply update the values in the mutable `stack_frames` that drive the operation. Unless I'm missing something, I don't believe this is suitable for `map/flat_map`

Reply:
Ah, I maybe I was just tired. I had some trouble understanding the code ^^